api/v1beta1/types.go

```diff
@@ -635,6 +635,24 @@ type ServiceEndpointSpec struct {
 	Locations []string `json:"locations"`
 }
 
+// NetworkInterface defines a network interface.
+type NetworkInterface struct {
+	// SubnetName specifies the subnet in which the new network interface will be placed.
+	SubnetName string `json:"subnetName,omitempty"`
+
+	// PrivateIPConfigs specifies the number of private IP addresses to attach to the interface.
+	// Defaults to 1 if not specified.
```

Comment:
We could, though I hesitate to recommend that given the real limit is imposed by the cloud provider. We can certainly mirror the constraint of a maximum of 256 private IP Addresses from Azure, but then if it changes in Azure, it will need to be updated in CAPZ as well. Do we typically mirror Azure API limits in the CAPZ code? I had the impression we usually just pass it through to the Azure APIs and let the Azure APIs enforce the limits.

Reply:
Makes sense to me. Would like to get @CecileRobertMichon thoughts too!

Reply:
In some cases it makes sense to enforce Azure limits for things that are static and not susceptible to change (eg. naming) to accelerate the feedback loop but in this case it does seem likely that Azure will change/increase the max IP limit in the future so I'm +1 on deferring to Azure API for validation